rpc/eth_api.go

```diff
@@ -313,13 +317,78 @@ func (e *PublicEthAPI) GetBlockByHash(hash common.Hash, fullTx bool) map[string]
 }
 
 // GetBlockByNumber returns the block identified by number.
-func (e *PublicEthAPI) GetBlockByNumber(blockNum rpc.BlockNumber, fullTx bool) map[string]interface{} {
-	return nil
+func (e *PublicEthAPI) GetBlockByNumber(blockNum rpc.BlockNumber, fullTx bool) (map[string]interface{}, error) {
+	value := blockNum.Int64()
+	block, err := e.cliCtx.Client.Block(&value)
+	if err != nil {
+		return nil, err
+	}
+	header := block.BlockMeta.Header
+
+	genesis, err := e.cliCtx.Client.Genesis()
+	if err != nil {
+		return nil, err
+	}
+	gasLimit := genesis.Genesis.ConsensusParams.Block.MaxGas
```

Comment:
Perhaps we can store this somewhere so we don't have to recompute?

Reply:
Well, the value has to be retrieved somehow, I agree it could be more efficient than querying the genesis data, but what would an alternative be? The block max gas is set through cosmos and I don't think we have access to it. Or are you saying to cache this value on the server for any future requests?